src/bun.js/node/types.zig

```diff
@@ -1835,548 +1849,2876 @@ pub const Emitter = struct {
+        return if (@TypeOf(T) != u16) ParsedPathUnion{ .u8 = result } else ParsedPathUnion{ .u16 = result };
+    }
 
-    pub fn create(globalObject: *JSC.JSGlobalObject, isWindows: bool) callconv(.C) JSC.JSValue {
-        return shim.cppFn("create", .{ globalObject, isWindows });
+    /// Taken from Zig 0.11.0 zig/src/resinator/rc.zig
+    /// https://github.com/ziglang/zig/blob/776cd673f206099012d789fd5d05d49dd72b9faa/src/resinator/rc.zig#L266
+    ///
+    /// Compares ASCII values case-insensitively, non-ASCII values are compared directly
+    fn eqlIgnoreCaseImpl(comptime T: type, a: []const T, b: []const T) bool {
+        if (@TypeOf(T) != u16) {
+            return std.ascii.eqlIgnoreCase(a, b);
+        }
+        if (a.len != b.len) return false;
+        for (a, b) |a_c, b_c| {
+            if (a_c < 128) {
+                if (std.ascii.toLower(@intCast(a_c)) != std.ascii.toLower(@intCast(b_c))) return false;
+            } else {
+                if (a_c != b_c) return false;
+            }
+        }
+        return true;
     }
 
-    pub fn basename(globalThis: *JSC.JSGlobalObject, isWindows: bool, args_ptr: [*]JSC.JSValue, args_len: u16) callconv(.C) JSC.JSValue {
-        if (comptime is_bindgen) return JSC.JSValue.jsUndefined();
-        if (args_len == 0) {
-            return JSC.toInvalidArguments("path is required", .{}, globalThis);
+    /// Based on Node v21.6.1 private helper formatExt:
+    /// https://github.com/nodejs/node/blob/6ae20aa63de78294b18d5015481485b7cd8fbb60/lib/path.js#L130C10-L130C19
+    inline fn formatExtImpl(comptime T: type, ext: []const T, buf: []T) []const T {
+        const len = ext.len;
+        if (len == 0) {
+            return "";
         }
-        var stack_fallback = std.heap.stackFallback(4096, JSC.getAllocator(globalThis));
-        const allocator = stack_fallback.get();
+        if (ext[0] == CHAR_DOT) {
+            return ext;
+        }
+        const bufSize = len + 1;
+        buf[0] = CHAR_DOT;
+        @memcpy(buf[1 ..bufSize], ext);
+        return buf[0 ..bufSize];
+    }
 
-        var arguments: []JSC.JSValue = args_ptr[0..args_len];
-        var path = arguments[0].toSlice(globalThis, allocator);
+    inline fn getCwd(buf: []u8) []const u8 {
+        const len = buf.len;
+        // Adopt the C idiom of indicating a fresh buffer
+        // with a null terminator since we're using Zig's std.c
+        buf[0] = 0;
+        // TODO: Handle error.
+        const res = std.c.getcwd(buf.ptr, buf.len);
+        return if (res != null) std.mem.sliceTo(res.?[0 ..len], 0) else "";
+    }
 
-        defer path.deinit();
-        var extname_ = if (args_len > 1) arguments[1].toSlice(globalThis, allocator) else JSC.ZigString.Slice.empty;
-        defer extname_.deinit();
+    /// Caller must free returned memory.
+    fn getBufAlloc(allocator: std.mem.Allocator, len: usize) ![]u8 {
```

Comment:
i dont think we should have this function at all

- the temporary buffers can be the global variables as already set, there is space reserved for three `PathBuffer` struct
- the 
- paths longer than MAX_PATH_BYTES can remain unsupported; (are there examples of JS libraries that depend on `node:path` operating beyond the max limit)

problems with this approach in particular are mostly around how it does not actually handle cleanup in any way. this will leak memory because the temp buffers are allocated but not freed in the success case. but i dont think we should ever even be up above MAX_PATH_BYTES and hit the inner branch.

since it is possible for a user to input a string longer than the max path length, i think we should just make this an error since the path these apis create will not be valid for use anyways.

Reply:
> paths longer than MAX_PATH_BYTES can remain unsupported; (are there examples of JS libraries that depend on node:path operating beyond the max limit)

We currently support paths longer than `MAX_PATH_BYTES`:
https://github.com/oven-sh/bun/blob/d16ac873476f79aef717473faa7a4b418152f110/test/js/node/path/path.test.js#L293-L304

This just covers our bases in other APIs. Node is in JS land where long-ish strings don't cause segfaults.
The cleanup is something to consider. My current take is if they are in a Big Data situation memory use will be high. At the moment it is a one-way pressure relief value (it expands).